### PR TITLE
Implement historical returnValue attribute for Event interface

### LINKED ISSUE #22881: Implement historical returnValue attribute for Event interface
https://dom.spec.whatwg.org/#event has a `returnValue` attribute which we should implement. This requires following the [spec](https://dom.spec.whatwg.org/#dom-event-returnvalue) for setting and getting the value, which should affect the `canceled` member of Event.

Code: `components/script/dom/webidls/Event.webidl`, `components/script/dom/event.rs`
Tests: `./mach test-wpt tests/wpt/web-platform-tests/dom/event` and then [update the expected results](https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations).

### PR DESCRIPTION
**Pull request by pngai**
Implementation of `returnValue` attribute for Event interface.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #22881 (GitHub issue number if applicable)

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because ___

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22933)

### CODE REVIEW

File: components/script/dom/event.rs [hunk 1]
```diff
@@ -297,6 +297,18 @@ impl EventMethods for Event {
...
+    // https://dom.spec.whatwg.org/#dom-event-returnvalue
+    fn ReturnValue(&self) -> bool {
+        self.canceled.get() == EventDefault::Allowed
+    }
+
+    // https://dom.spec.whatwg.org/#dom-event-returnvalue
+    fn SetReturnValue(&self, val: bool) {
+        if val == false && self.cancelable.get() {
+            self.canceled.set(EventDefault::Prevented)
+        }
```
Comment (nox) [inline components/script/dom/event.rs]: I think this would be cleaner:

```rust
if !val {
    self.PreventDefault();
}
```

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /dom/interfaces.html?exclude=Node:
  └ PASS [expected FAIL] Event interface: attribute returnValue

  ▶ Unexpected subtest result in /dom/interfaces.html?exclude=Node:
  └ PASS [expected FAIL] Event interface: document.createEvent("Event") must inherit property "returnValue" with the proper type

  ▶ Unexpected subtest result in /dom/interfaces.html?exclude=Node:
  └ PASS [expected FAIL] Event interface: new Event("foo") must inherit property "returnValue" with the proper type

  ▶ Unexpected subtest result in /dom/interfaces.html?exclude=Node:
  └ PASS [expected FAIL] Event interface: new CustomEvent("foo") must inherit property "returnValue" with the proper type
```

**Comment by jdm** [comment]
One more round of changes to ini files should do it.

**Comment by jdm** [comment]
@bors-servo retry
* #22955

**Pull request merged**